src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -836,6 +856,44 @@ private async Task<DownloadedBuild> GatherDropForBuildAsync(Build build, string
                             clientThrottle.Release();
                         }
                     }));
+
+                    // Now download the extras (if any)
+                    if (mustDownloadAssets.Any())
+                    {
+                        string extraAssetsDirectory = Path.Join(rootOutputDirectory, "extra-assets");
```

Comment:
A lot of this is duplicated from the code above. Can it be pulled out into a helper function so there's less duplicated code?

Reply:
Yeah you're right it just looked annoying to do at the time.  Refactored this block into a new `DownloadAssetsToDirectories()` function.